test/library/draft/Vector/Vector.chpl

```diff
@@ -906,26 +926,10 @@ module Vector {
       on this {
         _enter();
 
-        //
-        // TODO: This is not ideal, but the Sort API needs to be adjusted
-        // before we can sort over vectors directly.
-        //
         if _size > 1 {
-
-          // Copy current vector contents into an array.
-          var arr: [0..#_size] eltType;
-          for i in 0..#_size do
-            arr[i] = this[i];
-
-          Sort.sort(arr, comparator);
-
-          // This is equivalent to the clear routine.
-          _size = 0;
-          _capacity = _initialCapacity;
-
-          _requestCapacity(arr.size);
-          _extendGeneric(arr);
-          _maybeDecreaseCapacity();
+          _domain = {0..#_size};
+          Sort.sort(_data, comparator);
+          _domain = {0..#_capacity};
```

Comment:
I wonder if it would be better to do `Sort.sort(_data[0..#_size], comparator)`, rather than resizing the domain twice.  The slice isn't free, but neither is resizing the domain.

Reply:
Slice sounds better to me, though I don't know about the performance difference between two methods.

Reply:
https://github.com/chapel-lang/chapel/pull/16048/commits/6f9b6097c788666b4da98d7d9b428738eaf31946